app/controllers/users_controller.erb

```diff
@@ -0,0 +1,27 @@
+class UserController < ApplicationController
+
+    def new
+        @user = User.new
+    end
+
+    def create
+        @user = User.new(user_params)
+        if @user.save
+            session[:user_id] = @user.id
+            redirect_to user_path(@user)
+        else
+            render :new
+        end
+    end
+
+    def show
+        @user = User.find_by(id: params[:id])

```

Comment:
Same question, is there a risk of id being nil?

Reply:
In case of that scenario, it could be revised with

def show
    @user = User.find_by(id: params[:id])
    if @user.nil?
        # Handle the case where the user is not found, e.g., redirect to an error page
    else
        # Render the user profile page
    end
end